xnought/venome
# Frontend: Hide Elements based on Login

https://github.com/xnought/venome/assets/45224464/f10ba677-0f24-4b6f-98e1-9c6c65c069e5

### How to test

Since there are new dependencies, you must do the soft_restart shell command to install and refresh the packages. To test the changes, you should log into to the website with email "test" and password "test".

### What this PR does

- Stores the user's login status in a svelte store, allowing the website to keep track of whether the user is logged in
- Automatically checks if cookie exists when browser is refreshed; If it does, sets svelte store to login
- Deletes authentication cookie if user visits login page (e.g. if user logs out), and also clears stored login status
- Changes login button text between "login" and "logout" depending on user login status
- Hides / shows upload button based on user login status
- Hides / shows edit button based on user login status
- Automatically alerts the user and redirects them to home if they manually type in the URLs to the upload or edit pages
- Changed the test user email and password both to "test", to make it easier to test.

### Not addressed in this PR, but still needs to be done

- Frontend needs to be changed to send authentication tokens to backend from the cookie
- Backend needs to be changed to verify the authentication tokens prior to executing restricted requests (POST, PATCH, DELETE)
- Unauthorized access to upload/edit pages should redirect to a 401 Unauthorized page (not yet implemented) instead of home
- Some notification element should be created to better communicate to the user that they are logged in
- We should probably create some sort of basic user profile (so users can know which account they are logged into)

### New Dependencies

- svelte-store

Closes #165
Closes #166
